bioio/writers/ome_zarr_writer.py

```diff
@@ -1,431 +1,478 @@
         ----------
-        uri: biob.types.PathLike
-            The URI or local path for where to save the data.
+        store : Union[str, zarr.storage.StoreLike]
+            Path or Zarr Store for output.
+        shape : Tuple[int, int, int, int, int]
+            Base image shape (T, C, Z, Y, X).
+        dtype : Union[np.dtype, str]
+            NumPy dtype of the image data.
+        axes_names : List[str]
+            Axis names.
+        axes_types : List[str]
+            Axis types.
+        axes_units : List[Optional[str]]
+            Axis units.
+        axes_scale : List[float]
+            Physical scale per axis.
+        scale_factors : Tuple[int, int, int, int, int]
+            Downsampling factors per axis.
+        num_levels : Optional[int]
+            Maximum number of pyramid levels.
+        chunks : Union[str, Tuple[int, ...], List[Tuple[int, ...]]]
+            Chunk specification or "auto".
+        shards : Optional[Union[Tuple[int, ...], List[Tuple[int, ...]]]]
+            Shard specification.
+        compressor : Optional[BloscCodec]
+            Compressor codec to use.
+        image_name : str
+            Name for multiscale metadata.
+        channel_names : Optional[List[str]]
+            Channel labels.
+        channel_colors : Optional[List[str]]
+            Channel hex colors.
+        creator_info : Optional[dict]
+            Creator metadata.
         """
-        # Resolve final destination
-        fs, path = biob.io.pathlike_to_fs(uri)
+        self.shape = tuple(shape)
+        self.dtype = np.dtype(dtype)
+        self.axes_scale = axes_scale
+        self.scale_factors = scale_factors
+        # Multiscale shapes
+        self.level_shapes = self._compute_levels(num_levels)
+        # Chunks & shards
+        self.chunks = self._prepare_parameter(chunks, self._suggest_chunks, "chunks")
+        self.shards = self._prepare_parameter(
+            shards, lambda s: s, "shards", required=False
+        )
+        # Zarr store & arrays
+        self.root = self._init_store(store)
+        self.datasets = self._create_arrays(compressor)
+
+        # Build metadata
+        axes_meta = default_axes(axes_names, axes_types, axes_units)
+        channel_meta = None
+        if channel_names or channel_colors:
+            C = shape[1]
+            channel_meta = []
+            default_colors = [
+                "FF0000",
+                "00FF00",
+                "0000FF",
+                "FFFF00",
+                "FF00FF",
+                "00FFFF",
+            ]
+            for i in range(C):
+                color = (
+                    channel_colors[i]
+                    if channel_colors and i < len(channel_colors)
+                    else default_colors[i % len(default_colors)]
+                )
+                label = (
+                    channel_names[i]
+                    if channel_names and i < len(channel_names)
+                    else f"Channel {i}"
```

Comment:
Is this the OME default naming convention for channel names? I thought it was something like `channel:{index}` but idk could be totally misremembering